python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/run_handler.py

```diff
@@ -122,9 +122,39 @@ def to_default_asset_events(
                 project=None,
             )
 
-            if resource_type in REFABLE_NODE_TYPES and is_successful and not is_ephemeral:
+            if (
+                resource_type in REFABLE_NODE_TYPES
+                and result_status == NodeStatus.Success
+                and not is_ephemeral
+            ):
                 spec = asset_specs[0]
                 yield AssetMaterialization(
                     asset_key=spec.key,
                     metadata=default_metadata,
                 )
+            elif resource_type == NodeType.Test and result_status == NodeStatus.Pass:
+                metadata = {
+                    **default_metadata,
+                    "status": result_status,
+                }
+                if result["failures"] is not None:
+                    metadata["dagster_dbt/failed_row_count"] = result["failures"]
```

Comment:
is this something we already do in dagster_dbt? Just curious

Reply:
Yeah, we do the same thing for [dbt CLI events](https://github.com/dagster-io/dagster/blob/0714e927df01996d1ae6da3cf352b3aec1893555/python_modules/libraries/dagster-dbt/dagster_dbt/core/dbt_cli_event.py#L417-L418)